test/platforms.jl

```diff
@@ -111,3 +111,11 @@ using BinaryBuilderBase: get_march_flags, get_all_arch_names, get_all_march_name
     @test sort(get_all_march_names("x86_64")) == ["avx", "avx2", "avx512", "x86_64"]
     @test sort(get_all_march_names("armv7l")) == ["armv7l", "neonvfpv4"]
 end
+
+@testset "parse" begin
+    @test parse_platform("host") == HostPlatform()
+    @test parse_platform("any") == AnyPlatform()
+    @test parse_platform("x86_64-linux") == Platform("x86_64", "linux")
+    @test parse_platform("x86_64-linux-musl") == Platform("x86_64", "linux", libc="musl")
+    @test parse_platform("x86_64-linux-musl-gcc7") == Platform("x86_64", "linux", libc="musl", libgfortran_version=v"4.0.0")
```

Comment:
We shouldn't really be using the gcc stuff like that nowadays, I'm even surprised that works at all

Reply:
TBH this was Copilot's idea, with some proofing :P. I guess the AI is pretty good at these things.

Reply:
That's mildly terrifying